pkg/shared/util/version_mapping.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"github.com/numaproj/numaflow-go/pkg/info"
+)
+
+type sdkConstraints map[info.Language]string
+
+var versionMappingConfig = map[string]sdkConstraints{
```

Comment:
why do we need to maintain history mappings?

Reply:
The only benefit is that we can link this to the user-guide so that users can view history mappings.

Reply:
we are ok with just maintaining the mapping for the current version as well if that is perferred

Reply:
If that's the case, we do not need to maintain the history here - it might be too verbose to have everything including `-rc`. For documentation purpose, we should have well maintained doc for version compatibilities.

Reply:
ok we can add the historical mappings to release.md or create a separate yaml file in the root directory

Reply:
> ok we can add the historical mappings to release.md or create a separate yaml file in the root directory

This could be a separated task.